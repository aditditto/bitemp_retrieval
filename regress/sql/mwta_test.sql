DROP EXTENSION bitemp_retrieval;
CREATE EXTENSION bitemp_retrieval;
SET DATESTYLE TO 'ISO';

SELECT * FROM bitemporal_internal.ll_create_bitemporal_table(
    'public',
    'checkout',
    $$custid varchar(4) not null, 
      tapenum varchar(5) not null,
      cost integer
    $$,
   'tapenum');

SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.checkout',
    $$custid, tapenum, cost$$,
    $$'C101', 'T1234', 4$$,
    temporal_relationships.timeperiod('2020-06-01'::timestamptz, '2020-06-04'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);

SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.checkout',
    $$custid, tapenum, cost$$,
    $$'C102', 'T1245', 2$$,
    temporal_relationships.timeperiod('2020-06-03'::timestamptz, '2020-06-06'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);

SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.checkout',
    $$custid, tapenum, cost$$,
    $$'C102', 'T1234', 4$$,
    temporal_relationships.timeperiod('2020-06-07'::timestamptz, '2020-06-11'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);

SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.checkout',
    $$custid, tapenum, cost$$,
    $$'C102', 'T1245', 2$$,
    temporal_relationships.timeperiod('2020-06-17'::timestamptz, '2020-06-19'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);

SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.checkout',
    $$custid, tapenum, cost$$,
    $$'C102', 'T1245', 2$$,
    temporal_relationships.timeperiod('2020-06-19'::timestamptz, '2020-06-21'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);

SELECT custid, tapenum, cost, effective FROM checkout;

SELECT * FROM tmda_ci_now(
    'public',
    'checkout',
    '{}',
    ARRAY['count'],
    ARRAY['custid'],
    ARRAY['count']
) AS (
    count numeric,
    effective tstzrange
) ORDER BY lower(effective);

SELECT * FROM mwta_now(
    'public',
    'checkout',
    '{}',
    ARRAY['count'],
    ARRAY['custid'],
    ARRAY['count'],
    '2 days'::interval
) AS (
    count numeric,
    effective tstzrange
) ORDER BY lower(effective);

SELECT * FROM tmda_fi_now(
    'public',
    'checkout',
    '{}',
    ARRAY['count'],
    ARRAY['custid'],
    ARRAY['count'],
    '1 week'::interval
) AS (
    count numeric,
    effective tstzrange
);

DROP TABLE checkout;