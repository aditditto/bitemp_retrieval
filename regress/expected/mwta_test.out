DROP EXTENSION bitemp_retrieval;
CREATE EXTENSION bitemp_retrieval;
SET DATESTYLE TO 'ISO';
SELECT * FROM bitemporal_internal.ll_create_bitemporal_table(
    'public',
    'checkout',
    $$custid varchar(4) not null, 
      tapenum varchar(5) not null,
      cost integer
    $$,
   'tapenum');
 ll_create_bitemporal_table 
----------------------------
 t
(1 row)

SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.checkout',
    $$custid, tapenum, cost$$,
    $$'C101', 'T1234', 4$$,
    temporal_relationships.timeperiod('1970-01-01'::timestamptz, '1970-01-04'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);
 ll_bitemporal_insert 
----------------------
                    1
(1 row)

SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.checkout',
    $$custid, tapenum, cost$$,
    $$'C102', 'T1245', 2$$,
    temporal_relationships.timeperiod('1970-01-03'::timestamptz, '1970-01-06'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);
 ll_bitemporal_insert 
----------------------
                    1
(1 row)

SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.checkout',
    $$custid, tapenum, cost$$,
    $$'C102', 'T1234', 4$$,
    temporal_relationships.timeperiod('1970-01-07'::timestamptz, '1970-01-11'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);
 ll_bitemporal_insert 
----------------------
                    1
(1 row)

SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.checkout',
    $$custid, tapenum, cost$$,
    $$'C102', 'T1245', 2$$,
    temporal_relationships.timeperiod('1970-01-17'::timestamptz, '1970-01-19'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);
 ll_bitemporal_insert 
----------------------
                    1
(1 row)

SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.checkout',
    $$custid, tapenum, cost$$,
    $$'C102', 'T1245', 2$$,
    temporal_relationships.timeperiod('1970-01-19'::timestamptz, '1970-01-21'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);
 ll_bitemporal_insert 
----------------------
                    1
(1 row)

SELECT custid, tapenum, cost, effective FROM checkout;
 custid | tapenum | cost |                      effective                      
--------+---------+------+-----------------------------------------------------
 C101   | T1234   |    4 | ["1970-01-01 00:00:00-08","1970-01-04 00:00:00-08")
 C102   | T1245   |    2 | ["1970-01-03 00:00:00-08","1970-01-06 00:00:00-08")
 C102   | T1234   |    4 | ["1970-01-07 00:00:00-08","1970-01-11 00:00:00-08")
 C102   | T1245   |    2 | ["1970-01-17 00:00:00-08","1970-01-19 00:00:00-08")
 C102   | T1245   |    2 | ["1970-01-19 00:00:00-08","1970-01-21 00:00:00-08")
(5 rows)

SELECT * FROM tmda_ci_now(
    'public',
    'checkout',
    '{}',
    ARRAY['count'],
    ARRAY['custid'],
    ARRAY['count']
) AS (
    count numeric,
    effective tstzrange
) ORDER BY lower(effective);
 count |                      effective                      
-------+-----------------------------------------------------
     1 | ["1970-01-01 00:00:00-08","1970-01-03 00:00:00-08")
     2 | ["1970-01-03 00:00:00-08","1970-01-04 00:00:00-08")
     1 | ["1970-01-04 00:00:00-08","1970-01-06 00:00:00-08")
     1 | ["1970-01-07 00:00:00-08","1970-01-11 00:00:00-08")
     1 | ["1970-01-17 00:00:00-08","1970-01-19 00:00:00-08")
     1 | ["1970-01-19 00:00:00-08","1970-01-21 00:00:00-08")
(6 rows)

SELECT * FROM mwta_now(
    'public',
    'checkout',
    '{}',
    ARRAY['count'],
    ARRAY['custid'],
    ARRAY['count'],
    '2 days'::interval
) AS (
    count numeric,
    effective tstzrange
) ORDER BY lower(effective);
 count |                      effective                      
-------+-----------------------------------------------------
     1 | ["1970-01-01 00:00:00-08","1970-01-03 00:00:00-08")
     2 | ["1970-01-03 00:00:00-08","1970-01-06 00:00:00-08")
     1 | ["1970-01-06 00:00:00-08","1970-01-07 00:00:00-08")
     2 | ["1970-01-07 00:00:00-08","1970-01-08 00:00:00-08")
     1 | ["1970-01-08 00:00:00-08","1970-01-13 00:00:00-08")
     1 | ["1970-01-17 00:00:00-08","1970-01-19 00:00:00-08")
     2 | ["1970-01-19 00:00:00-08","1970-01-21 00:00:00-08")
     1 | ["1970-01-21 00:00:00-08","1970-01-23 00:00:00-08")
(8 rows)

DROP TABLE checkout;
